### PR TITLE
Revert code to original submission

### PR DESCRIPTION
**Pull request by kntsoriano**

### CODE REVIEW

File: app/javascript/components/Editor.tsx [hunk 1]
```diff
@@ -30,6 +30,8 @@ import { RunTestsButton } from './editor/RunTestsButton'
 import { SubmitButton } from './editor/SubmitButton'
 import { useIsMounted } from 'use-is-mounted'
 import { camelizeKeys } from 'humps'
+import { useSaveFiles } from './editor/file-editor/useSaveFiles'
+import { isEqual } from 'lodash'
```
Comment (kntsoriano) [inline app/javascript/components/Editor.tsx:34]: We need this to perform deep equality between the current `files` array and the initial `files` array passed.

File: app/javascript/components/Editor.tsx [hunk 2]
```diff
@@ -313,7 +322,10 @@ export function Editor({
             setKeybindings={setKeybindings}
             setWrap={setWrap}
           />
-          <Header.ActionMore />
+          <Header.ActionMore
+            onRevert={revertContent}
+            isRevertDisabled={isEqual(initialFiles, files)}
```
Comment (kntsoriano) [inline app/javascript/components/Editor.tsx:358]: We disable the revert button if the files haven't changed.

File: app/javascript/components/editor/FileEditor.tsx [hunk 3]
```diff
@@ -48,58 +36,6 @@ export function FileEditor({
   files: File[]
   wrap: WrapSetting
 }): JSX.Element {
-  // const languageServerUrl: string = useMemo(() => {
```
Comment (kntsoriano) [inline app/javascript/components/editor/FileEditor.tsx:51]: I've extracted this to a custom hook in order for us to not comment this code out anymore.

File: app/javascript/components/editor/file-editor/useSaveFiles.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React, { useEffect } from 'react'
```
Comment (kntsoriano) [inline app/javascript/components/editor/file-editor/useSaveFiles.ts]: This hook periodically saves the files to local storage every 500 ms.

File: app/javascript/components/editor/header/usePanel.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState, useRef } from 'react'
```
Comment (kntsoriano) [inline app/javascript/components/editor/header/usePanel.tsx:1]: This hook allows us to use show / hide a panel when clicking on a specific button. This also closes the panel when the user clicks outside it.

File: app/javascript/utils/use-storage.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import React, { useCallback, useState } from 'react'
+import React, { useState } from 'react'
```
Comment (kntsoriano) [inline app/javascript/utils/use-storage.ts:1]: This handles saving to local storage. We need to encode / decode the objects to JSON as localstorage only accepts strings.

File: app/javascript/components/editor/FileEditor.tsx (hunk 3 above)
Comment (ErikSchierboom) [reply app/javascript/components/editor/FileEditor.tsx:51]: Great!

File: app/javascript/components/editor/header/Settings.tsx [hunk 7]
```diff
@@ -18,43 +18,15 @@ export function Settings({
...
-  }, [open])
+  const {
+    open,
+    setOpen,
+    buttonRef,
+    panelRef,
+    componentRef,
+    styles,
+    attributes,
+  } = usePanel()
```
Comment (ErikSchierboom) [inline app/javascript/components/editor/header/Settings.tsx:29]: I really like how this `usePanel` hooks simplifies things!

File: app/javascript/components/editor/header/usePanel.tsx (hunk 5 above)
Comment (ErikSchierboom) [reply app/javascript/components/editor/header/usePanel.tsx:1]: Excellent work!

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**